### ISSUE TITLE
Adjusting the economic model

### ISSUE THREAD

**Issue by erikzhang** [neo-project/neo#285]
Adjusting the economic model for gas distribution.

Agreed:

- Allow gas to use the decimal part when being used as sys_fee

Contentious:

- A small increase in the supply of gas every year
- Distributing GAS only to the voting NEO
- Reward consensus nodes

**Comment by igormcoelho** [comment]
This one I don't get the direction yet... any ideas brother @vncoelho?

**Comment by vncoelho** [comment]
A naice veri gud.
I think that is could be crossed in time, ma broda, not in terms of blocks. @lerider

If the generation is expected 8 GAS in terms of 15s it should be 0,54 per 1000ms.

In this sense, Ontology and other private blockchains could easily follow this formula and generate GAS accordingly. This would unify the price of gas across different blockchains ecosystems.

**Comment by deanpress** [comment]
If the economic model is moving to `proof of x` + infinite GAS supply, wouldn't it become more similar to Ethereum Casper except there are more complexities due to dual-token model (which was not designed for this model) and centrally decided upon fee structure?

One of NEO's main attracting powers was always that if decentralization would be introduced, consensus nodes can be run by partners without direct incentives from transaction fees. Consensus nodes would benefit from other factors such as funding/exposure/contributing to ecosystem etc, and the blockchain fees would be distributed only to NEO holders. NEO holders would also be in control over the fees.

The game theory behind it is interesting refreshing and it hasn't even proven itself right or wrong yet, so why are there already considerations for moving to a `proof of x` model?

**Comment by erikzhang** [comment]
> If the economic model is moving to proof of x + infinite GAS supply, wouldn't it become more similar to Ethereum Casper except with more complexities due to dual-token model (which was not designed for this model) + centrally decided upon fee structure (instead of letting NEO holders vote)?

In the dual-token model, if neo enters a blackhole address, it causes all of the gas that this part of neo gets lost, and the process, like a real blackhole, will eventually devour all the gas.

**Comment by lerider** [comment]
**A small increase in the supply of gas every year**
**Reward consensus nodes**
I strongly oppose these. The economic model is built upon circulation in a closed system, aligning interest so that both users and CN's have the same motives. Allowing increase in GAS based on for example reward to CN's nodes will create a system where CN's main interest is of economic gain and not for the benefit of users. It will result in a consortium of CN's who manipulate GAS price for their own maximized profit. It may potentially give a short term spike in NEO price when people are hoarding voting power (see EOS), but once hoarding is complete, the CN's will gain most benefit from "stabilizing" the consortium and we are stuck with CN's that never change owner (unless they dump on purpose). There are already dozens of parties who want to carry the cost of a CN (around 500 USD a month) without direct economic gain; I do not see a reason to overhaul the whole economic model.

**Make neo divisible**
The only thing this would help is to allow distribution of smaller GAS fees. The small benefit does not justify the big risk forced upon token holders.
1. NEO is the governance token and GAS is the utility token. The characteristics of the tokens support this; if you want to fuel a contract with a system asset, then you use the utility token (GAS). This will increase demand on GAS token as the network usage increases, and in turn increase value of NEO when GAS is recycles. Allowing NEO to be divisible create a big risk that dApps choose to fuel themselves with NEO token instead. This may be negative in terms of token price for BOTH NEO and GAS token; as GAS loses demand, the value decrease. As the recycling benefit decreases in value, NEO token may also lose value. This is my personal expectation, and I do not see the big risk for token holders justify the small benefit.
2. Even though high voting participation is beneficial, we would like to avoid random voting as much as possible. Allowing fractional NEO to vote (for reward) will give much power to wallet creators, as they will certainly implement a "vote standard" function (for their own nodes?) to make it easier to vote for users. People who are more invested will have more incentive to vote "properly", and keeping the NEO as indivisible is at least a small barrier for participation in voting (for good and bad).

**Allow gas to use the decimal part when being used as sys_fee**
I think the "decimal part" refers to decimal part of the NEO token when distributing GAS so that smaller fees can be accepted. I greatly support smaller fees, but think we should try to find another way to allow the recycling rather than making NEO divisible.

In summary, I do not support these changes. It has too much influence of the economic model and there is no reason big enough that justify these changes. Moreover, it has too much impact on both developers and token holders; it creates an uncertainty. Developers and token holders need to know what they can expect in the future, and this overhaul is too big without enough reason to do so.

**Comment by erikzhang** [comment]
@lerider Please see my reason at https://github.com/neo-project/neo/issues/285#issuecomment-403859121

**Comment by Edgegasm** [comment]
I agree with @lerider that incentivizing consensus nodes & increasing GAS supply should not be implemented. It disturbs the closed economic model that NEO was created to serve. I also have qualms about divisible NEO, as this is detrimental to the value of GAS and use of a two-token model, although I understand that it is ideal for accessibility as NEO's price increases.

My suggestion about an improved voted mechanism using time-locked voting contracts may prove a useful solution to this issue. It also suggests approval voting, which should drastically reduce the risk of random voting whilst also minimizing the potential vote influence of centralized exchanges. It will also incentivize voting, and could prevent the risk of GAS being lost to a 'black hole' address as @erikzhang has described it.

Please view my proposal [here ](https://github.com/Edgegasm/proposals/blob/d5d71d91c82629c4968331d71969db2117193116/nep-11.mediawiki)for more information.

**Comment by BehindYou27** [comment]
I strongly oppose these changes. In my opinion, divisible Neo is NOT needed and serves no noticeable reason. 
In case you have a point to change this immediately, please point these out.

Gas-Inflation:

I’m not in favor of this at all. This would punish all gas-holders.
In addition (like Malcolm pointed out) CN’s don’t need the additional economical benefit. They will cover their expenses by transaction fees. This has to be enough to make it interesting for these parties.

To be completely candid: I don’t understand the reasoning behind these changes anyway. Neo is right now not that expensive that we would have a liquidity-issue. 
Gas price is not that expensive that a inflation is needed to prevent holding. 

Why change a running system?

**Comment by erikzhang** [comment]
@Edgegasm We don't need to worry about gas in the black hole address. But we do need to worry about neo in the black hole address. Because in the current economic model, the gas that is consumed will be given to neo holders. So gas will flow into the black hole address until all gas is gone.

**Comment by Edgegasm** [comment]
@erikzhang If GAS is only distributed to voting participants (incentivizes voting), and we use a contract approach as my suggestion, GAS would never flow into a black hole NEO address. The NEO at that address would not ever be placed into a voting contract, so the address would not be eligible for GAS distribution.

As a result, lost NEO would only serve to lower the circulating supply & increase the value for other NEO holders.

**Comment by erikzhang** [comment]
> If GAS is only distributed to voting participants (incentivizes voting), and we use a contract approach as my suggestion, GAS would never flow into a black hole NEO address.

Addresses that lose their private keys can also cause problems.

**Comment by Edgegasm** [comment]
In what way? Without access to the private key, you wouldn't be able to send your NEO to the voting contract. Without voting, no GAS will be distributed to the address.

**Comment by erikzhang** [comment]
I sent neo to voting contract, and then lost private key. So I can never withdraw them back. Does this cause problems?

**Comment by BehindYou27** [comment]
@erikzhang 

Can you point out what exactly will be the problem if Gas is locked up in dead wallets?
Is it only the appreciation of Gas?

On what bases would you like to have the inflation?
Some random number like EOS has? A calculated inflation?

There is one thing I could get behind: If the Gas is redistributed to the holders/voters. But it can’t be a random number. I’d propose that the lost gas has to be calculated over a long time. Maybe 22 years after the genesis block. 
So you would have to check what wallets didn’t touch the generated gas for 22 years and add this to the Gas-distribution. 

The longer the timeframe, the more secure this system becomes for holders. 

I still won’t like it but I could deal with it. That would mean in case someone would regain access to the gas that was locked up for 22 years in their wallet, the additional distribution would stop until we have less than 100 Million gas in realistic supply

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by t-ant8** [comment]
@lerider @deanpress I think, rewarding consensus nodes was planned from the beginning if I remember correctly. It was written in the first Antshares whitepaper (CN were called "book-keeper" nodes).

I think GAS-incentive makes it more attractive for large businesses (such as KPN) to run a CN node. In the end, running CN is just business for coorporates (IMO)

**Comment by Edgegasm** [comment]
@t-ant8 Rewarding consensus nodes was only intended to be done using transaction fees, with the idea that we can keep them low if not zero by competition for the right to participate in consensus.

**Comment by erikzhang** [comment]
@Edgegasm Can you put your solution in a separate issue?

**Comment by Edgegasm** [comment]
@erikzhang It's currently up as a [pull request](https://github.com/neo-project/proposals/pull/55). Would you like me to post it as an issue instead?

**Comment by erikzhang** [comment]
> It's currently up as a NEP pull request. Would you like me to post it as an issue instead?

Yes, please post it as an issue so that we can discuss it in the context of NEO 3.0

**Comment by BehindYou27** [comment]
Hi @t-ant8 

But what would be the reason behind this? I don’t think that we will have problems to find enough CNs that are willing to spend this money to run a node. 

So we could decide if they get a Gas-Incentive to run it, which would effectively take away value from every gas-holder due to increased supply 

Or

CN’s will try to make it efficient by having transaction fees. This would take away gas from people that are using the system. 

Which one is fairer? 

I know what I think.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Edgegasm** [comment]
@erikzhang Issue opened for discussion [here](https://github.com/neo-project/neo/issues/313).

**Comment by erikzhang** [comment]
In fact, I don't like to increase gas supply either. If the problem of black hole addresses can be solved, there is no reason to do so.

**Comment by Edgegasm** [comment]
Indeed, one of the most widely disliked parts about FIAT currency is inflation. It should be avoided at all costs.

**Comment by Edgegasm** [comment]
@t-ant8 Transactions fees (if enabled) go to CNs. System fees (contract deployment etc) are redistributed to NEO holders.

**Comment by ThomasLobker** [comment]
The black hole problem is not really a problem. Less NEO and GAS in circulation will simply increase the value and thus making it more likely for people to be more careful in the future. This is no different with Bitcoin, or even paper money. Stuff gets lost or breaks. I don't see how this is a problem.

**Comment by coranos** [comment]
It is my understanding, If the amount of neo in circulation decreases below 66% of total, you will be unable to reach consensus because you have 33% of your neo present but not voting.

So blackholed neo is a problem.

**Comment by Edgegasm** [comment]
@coranos Not a problem with the time-locked voting contract proposal as only the locked up votes are being counted. How much NEO is in circulation doesn't matter; all that matters in determining CNs is the vote distribution.

**Comment by cryptogirlHODL** [comment]
@coranos Please do look at @Edgegasm 's proposal! I think it solves a lot of issues, including this black hole issue. Additionally it removes voting power from exchanges, incentivizes people to hold in their private wallets (since exchanges cannot pay out GAS for them). As a fun side-bonus, it doubles the HODL power of the NEO token too since due to the lock-up less panic-selling can occur which can only be positive for price stability.

It's a multi-edged blade.

**Comment by blahd** [comment]
If GAS is lost to blackhole addresses is it worth doing a back of an envelope calculation to estimate how much utilization the network could handle before GAS reaches its divisibility limit for various conditions of e.g. rate of GAS loss, network load etc. 

Perhaps the rate of loss would turn out to be a non-issue if the chance of GAS reaching its divisibility limit is vastly lower than a conceivable high usage global model would require?

**Comment by ThomasLobker** [comment]
> If the amount of neo in circulation decreases below 66% of total, you will be unable to reach consensus because you have 33% of your neo present but not voting.

@coranos that would mean if people were just to lazy to vote, we will also be unable to reach consensus. I'm assuming that we will only look at the actual votes, not at sleeping (or blackholed) NEO.

**Comment by CrypJohn** [comment]
In the end we also will have to deal with inflation. The current system is good, but you will attrack more people with;
Make neo divisible
Allow gas to use the decimal part when being used as sys_fee

**Comment by t-ant8** [comment]
@coranos, i think the 66% relates to the number of CNs and not to the amount of NEO for consensus to be achieved (dBFT).

**Comment by BehindYou27** [comment]
@CrypJohn
 
>> you will attract more people with; Make Neo divisible

That is true. However in Germany we have a saying for this. It’s called “Milchmädchenrechnung” 
It’s a calculation that works but that doesn’t make sense. 

By making Neo divisible you are effectively increasing the supply. 
It would probably increase liquidity and lead to a short term increase in price. For the long term I see big risks for Gas, f.e. rendering it’s use case useless, because Neo could be used for everything then. 
If Gas loses value, so should Neo-because the value of Neo is tied to the value of Gas. (But because Neo takes over this part from Gas, it should balance itself out) 

I know that you are probably aware of this. I still wanted to point this out because we probably have visitors that may still learn about Neo. 

I also know that you didn’t say that you are in favor of this. You only stated that it will attract people. I agree. It does attract people but for the wrong reasons. 

It’s like an exchange listing that boots the price of a token. For a token that has low liquidity it makes sense. For a well established token, it does not. It’s not sustainable growth.

**Comment by Edgegasm** [comment]
This is why the time-lock contract approach makes so much sense. We want divisible NEO for the accessibility it provides. It will aid us in becoming decentralized, and it will make it more likely to stay that way.

NEO is scarce because it's indivisible. But if most NEO are going to be locked up for voting/getting GAS, then we create scarcity again. This method allows us to use divisible NEO *without* harming the value proposition for GAS as a utility token. Best of both worlds. Incentivized voting, NEO scarcity, and valuable GAS.

**Comment by CrypJohn** [comment]
@BehindYou27 

Agree with your feedback, deleted that part just after posting. To much negative effects.

**Comment by saltyskip** [comment]
I'd like to make an argument FOR inflationary gas which seems to be strongly opposed here. As @erikzhang  already, mentioned the black hole problem leads to a further and further decreasing supply of GAS. 

Any deflationary currency rewards hoarding instead of spending. If GAS is inflationary then it should increase the velocity of transactions inside the NEO network, which benefits NEO holders. Even if GAS is inflationary, then percentage of the total GAS that you own remains constant, assuming you do not change the amount of NEO you hold.

IMO in the two token role, NEO is meant for holding, and GAS is meant for spending. Making GAS inflationary actually strengthens this relationship rather than weakens it

**Comment by Murquia** [comment]
To prevent blackhole, shouldn't there be a default system recourse where an unused wallet gets deleted with its tokens redistributed to the rest of the active token holders or simply returned to the ecosystem after a certain duration of inactivity? (e.g. 10 years or.. 50 years?).

**Comment by BehindYou27** [comment]
@Murqua

Yes. This is a very crucial point. 

I’ve pointed out a proposal with my second comment below this post:

https://github.com/neo-project/neo/issues/313

**Comment by cryptoecon** [comment]
Since there are ongoing conflicting interests between NEO holders and GAS holders, whose views or proposals are basically biased, it is fair to stick to the token economic model set in the original whitepaper. A divisible NEO could increase the usage of NEO and be beneficial to the NEO holders, but it will be detrimental to the whole ecosystem in the long term. As for the black holes problem, I think it is not a big deal and we need to find other ways to fix it.  Definitely, it is not worthwhile to alter the basic framework of the NEO system for such a minor issue.

**Comment by Edgegasm** [comment]
> It will indeed be hard to buy 1 NEO for regular investors. Is this a bad thing however? 

I think the answer to this is unquestionably *yes*, it is a bad thing. We require decentralization as a high priority. Having 10000 people voting rather than 1000 is a net positive, even if some have higher vote power than others due to their own financial stakes. Of course, it doesn't make a difference if the vast majority of tokens are held by a select few, but as long as this is not the case, we can attain decentralization this way. We need to design a system that allows *anyone* to get involved, and then we need to start thinking about token distribution. Divisible NEO allows more people to join in on consensus, and 1 NEO = 1 Vote power allows financial stakes to be respected. This balance is ideal for the smart economy. 

It's all in the interests of diversifying where those votes are coming from. We cannot claim decentralization if only businesses/exchanges/whales/early adopters are controlling votes. It needs to be more open than this. In addition, using both indivisible NEO & time-locked voting will drive scarcity through the roof. NEO's price will become an obscene barrier, and management of the network will stagnate. We'll end up with top-heavy token distribution, resulting in consensus plutocracy.

> It will become scary to vote and see your expensive (4/5/?? digit NEO) temporarily being moved away from your wallet

A wallet can pull the contract hash and just list your NEO as 'locked for voting,' but as you've said, the wallets are essentially contracts already. If we can produce a special transaction locks tokens inside the existing wallet, registers specific candidate nominations & can track/count votes, then I agree that this is an ideal approach. I think that is going beyond the scope of a special transaction though. We shall have to defer to the devs. @erikzhang might have some thoughts.

They are however in effect the same thing. Contracts are used to leverage a time-locking mechanic and to allow a certain amount of calculation/validation. It doesn't make a difference to the system if NEO is sent to a new contract address, or if it would somehow remain locked under your own address. Either way, a transaction must still be registered. Without transfers, there is no data to go in a block. Without block data, the votes are not immutable. Indeed one will likely be preferable for the user experience, so we shall have to see if it's possible.

> This would also make it easier to cancel/modify votes, and would spam the network less.

Votes should not be cancellable (at least after the locking period). Prior the locking period, votes can simply be withdraw as in any other contract. The network will be spammed the same amount; as I mentioned earlier, vote transactions must be recorded in blocks in order for the voting system to remain immutable.

**Comment by toghrulmaharramov** [comment]
@erikzhang The blackhole problem could be solved by distributing GAS only to the voting NEO. Even if the private key of an address is lost after voting, the GAS would be distributed to the wallet only until the next voting period starts. Another solution would be to distribute GAS claims only with the present NEO, meaning that the GAS cannot be claimed after NEO has been moved out of the wallet. This would simplify the distribution scheme as well.

I agree with Erik that the nodes have to be incentivized more than they currently are. The platform should not rely on goodwill, this is a short-sighted approach.

**Comment by ThomasLobker** [comment]
> I think the answer to this is unquestionably yes, it is a bad thing. We require decentralization as a high priority. Having 10000 people voting rather than 1000 is a net positive, even if some have higher vote power than others due to their own financial stakes.

It's actually very unlikely that allowing more votes will change the outcome. Proper [data sampling](https://en.wikipedia.org/wiki/Statistics#Sampling) is very reliable. and you must also consider the following two facts:

- Not all NEO users are also NEO asset holders (voters)
- Not all NEO asset holders (voters) are also NEO users

So this should not be considered political voting. The users are not voting, the shareholders are voting. What's important is that the voting outcome represents the majority of the shareholders, weighted by how much NEO they are holding. If we have 100,000,000 samples or 1,000,000,000 samples makes no difference. Something like 10,000 samples will probably be more than enough. Mandatory [cumulative voting](https://en.wikipedia.org/wiki/Cumulative_voting) will help to prevent political and tactical voting too. We should require each NEO to be voting on three candidates, so that no candidate can gain more than 33% of the votes.

Locking is very unnecessary and not appealing at all.

I propose a very simple strategy:

- Any NEO address with NEO inside will be encouraged to make a vote. Only once you made your vote you will be able to get GAS assigned to your address, for you to claim
- Your vote requires a minimum of three candidates. Each candidate will receive its share of votes, based on the amount of NEO being held by the address voting. If you gain more NEO, your votes will gain more weight. If you transfer our NEO your votes will lose weight.
- Voting is not an event, but a state. If any candidate appears to be a bad actor, voters should be able to revoke their vote immediately. The state changes on each block.

This is actually a very light system. It's very easy to keep track of votes. Each NEO address will have a number of candidates that is voted on, multiplied by the amount of NEO in that address, divided by the amount of votes for that address. This can be recalculated on a ContractTransaction involving that NEO address. It's a fair and reliable system that doesn't require locking, staking or other tricks.

**Comment by toghrulmaharramov** [comment]
@ThomasLobker I think you're missing the point. You cannot equate the Consensus Node elections to a shareholder voting event, because in a corporate world, a single entity owning a 50%+1 stake is a norm, which cannot be said for NEO as it is geared towards decentralization. NEO voting mechanism should be considered to be a mix between political and shareholder voting as otherwise decentralization will not work.

**Comment by vncoelho** [comment]
Hi @erikzhang, could you update the post with the ideas more related to the current discussions?

According to our recent meetings with communities and developer, I believe that we most are not supporting:
* Increasing GAS supply
* Make NEO divisible

Thus, the proposal could be kept to:
* Allow gas to use the decimal part when being used as sys_fee
* Distributing GAS only to the voting NEO (maybe)
* Reward consensus nodes (maybe)

**Comment by EdgeDLT** [comment]
I'm not sure how we can reward consensus nodes without increasing GAS supply. We shouldn't keep letting them take transaction fees, this is a flawed model. It incentivizes the consensus node to drive up transaction fees (e.g. filling blocks with free, spam transactions to force users to pay network fees).

**Comment by coranos** [comment]
Since NEO will be at 7 consensus nodes forever, increasing consensus nodes slows down the network, I don't think consensus nodes should get any reward at all.

**Comment by vncoelho** [comment]
@EdgeDLT, I still believe in the initial idea that those maintaining the consensus are those interested in running it. Surely, there are many. 
In addition, more can come, for example, Universities using Smart Contracts services.
Thus, from my perspective, I do not really think that reward consensus node is a necessary reward.

@coranos, I will just discard your comment.

**Comment by coranos** [comment]
@vncoelho your comment is a better phrasing, so i discarded my comment.

**Comment by ThomasLobker** [comment]
> I'm not sure how we can reward consensus nodes without increasing GAS supply.

Each block unlocks 7 GAS at the moment. We could technically let a consensus node take a part of that? But I personally like that idea that the incentive for a consensus node is not directly to make money. Run a consensus node if you have a stake in the network, or just want to sponsor the network.

**Comment by EdgeDLT** [comment]
I agree @vncoelho, running a NEO consensus node can be its own reward. In a decentralized world, I can imagine reputation goes a long way. Helping safeguard and maintain the blockchain is a good way to build a strong relationship with that network's users (and the market).

**Comment by toghrulmaharramov** [comment]
Hey guys, just popped by to express my opinion. I believe that the approach to the consensus and the cryptoeconomics of the protocol is a bit skewed at the moment. The voting mechanism is yet to be designed and the underlying Proof-of-Stake protocol has no formal proofs (for example, the current iteration of dBFT is vulnerable to "long-range" attacks). I would start by formalizing the voting mechanism, followed by the definition of the Proof-of-Stake and ending with the incorporation of the aforementioned proofs into the dBFT formal proof. Only then you would be able to formalize the cryptoeconomics of the protocol, as otherwise, you're playing a guessing game.

Also, I vehemently disagree with an idea that the consensus nodes should not be monetarily incentivized as this contradicts the game theoretic assumptions of any given economic model, as well as being pro-plutocracy.

I apologize for having no time on my hand to make any concrete input, but I hope that my opinion helps.

**Comment by PBSJ** [comment]
@vncoelho - when you say "increasing GAS supply" are you talking about introducing an inflationary model? 

Would be curious to explore the pros and cons of that. 

One pro I can see would be that there would be an incentive to spend GAS. 

One con I can see is that it would support the capitalist status quo of eternal growth (which, however, may be exactly what is intended for NEO)

**Comment by EdgeDLT** [comment]
Hey @toghrulmaharramov, good to see you. Could you explain your thoughts on this part for me?

> as well as being pro-plutocracy.

I'm not sure how this can be true. Maybe from the perspective that those without funds to operate nodes can't break even, so there's a barrier to entry for noderunners?

We have seen PoS networks where validators receive large incentives for their participation. Since those awarded tokens are also used to vote in these networks, it seems logical to say that this act leads to plutocracy. The longer you are a node, the more voting power you gain to consolidate your rule. Naturally, this would result in cartels.

So, I'm not quite sure why the inverse could be true. Can you correct my thinking here if you have time?

**Comment by toghrulmaharramov** [comment]
Hey @EdgeDLT.

Lack of incentivizations for running a node leads decreases the number of users being able to afford to run the nodes and plutocracy as a result. Apart from that, the probability of various "bribery" attacks increases when the protocol does not include any incentivization mechanisms. Finally, the security assumptions will have to explicitly define non-Byzantine replicas as honest users rather than economically rational users.

No need to retain the same consensus committees for prolonged epochs, a much better idea is to create medium-length epochs and pseudo-randomly shuffle the committees every epoch (e.g. using Algorand-like VRF sortition or Snow White-like hash extraction).

**Comment by EdgeDLT** [comment]
> Lack of incentivizations for running a node leads to a decreased number of users being able to afford to run the nodes and plutocracy as a result. 

From what I figure about how many consensus nodes we will actually have, is this really an issue? The bare minimum cost for running and safeguarding a consensus node doesn't seem so insurmountable. I suppose that would change over time with chain size though.

The real risk of plutocracy really stems from the token distribution, which can't be easily prevented unless we end up all voting through digital identities in a fully digital democracy.

> Apart from that, the probability of various "bribery" attacks increases when the protocol does not include any incentivization mechanisms.

Seems like those incentivization mechanisms *are* the bribery attacks. Example, EOS BPs, Bitcoin/Ethereum pools, etc. Incentive causes centralization. People are always going to follow the money, and those who are best equipped to do so will reap the most rewards. 

I think it is better to detach this pursuit of incentive from block validation entirely. It would be much more interesting (and productive) to find a way to incentivize peers!

> No need to retain the same consensus committees for prolonged epochs

Agree with you there, I suppose it would allow us to increase our number of nodes.

**Comment by vncoelho** [comment]
@toghrulmaharramov, it is great to see you here again! :)

Tog, we have been investigating consensus intensively, but I still believe that the current design is quite adequate.
I do not like these ideas of shuffle and dividing the committees. Before doing so, I believe that we should explore more the possibilities of sharding and how the current design can be extended.

The main advance of the dBFT 2.0 is its simplicity, being (maybe) the only yet working consensus with 1 block finality able to support `f` faulty nodes. As you known, it  inspired on the pBFT and has a precisely done design for achieving all needs.

**About the voting**: Voting are working since NEO 2.x early versions and they are a key change on the Delegated strategy. NEO holders have the potential of eliminating CN that do not correspond to the desired needs.
We are currently doing a short-movie for didactically introducing the current system.

**Comment by toghrulmaharramov** [comment]
@EdgeDLT The costs of running the node will increase as the blockchain size increases. For example, Ethereum archival node sizes have already exceeded 1 TB.

Regarding the token distribution, as long as the reward model is designed in a way to incentivize rather than enrich the consensus participants (let's say you get an average annual return of 20% from running a consensus node), the outcome of such a model is highly unlikely to lead to a plutocratic system.

No, an example of a bribery attack would be a "long-range" attack, which EOS and Bitcoin are statistically and probabilistically resistant to respectively. Also, NEO is far more centralized than EOS, let alone Bitcoin/Ethereum (while the mining pool argument is a valid one, the permission-less nature of Proof-of-Work algorithms enables the "honest" (a loose definition in this case) participants of the mining pool to stop contributing their hashrate to the pool).

**Comment by toghrulmaharramov** [comment]
@vncoelho good to hear from you!
 
The issue is not in the design of the consensus protocol (though there's still room for improvement there, in my opinion), but in the design (or precisely lack thereof) of the underlying Proof-of-Stake protocol. From what I've seen in the Ethereum camp, we're still some time away from a viable sharding proposition (OmniLedger guys might want to argue with that), whereas committee shuffling techniques and reputation modules are already gaining widespread use (including in my own work).

Actually, that is not the case. Algorand and HoneyBadgerBFT have 1 block finality while being able to preserve liveliness and consistency under f Byzantine replicas (though Algorand's fault-tolerance depends on the predefined parameters), as well as DFinity having 2 block finality under f Byzantine replicas. You could also add Tendermint and Hyperledger Fabric to the list.

The issue with the current voting system is that does nothing to improve on the issues that were discussed almost a year ago, which is what I mean when saying that the voting mechanism is yet to be designed (the one currently in use is a trivial one, with all due respect).

**Comment by vncoelho** [comment]
@toghrulmaharramov, I understand your point, really.
I also would like to make it clear that I am not criticizing Ethereum (I just bought some more these days ago. I think that have their camp and a brilliant future as well), but I would like to say that we are strictly following Istanbul. 
I should say that, from my limited perspective, we are currently the pioneers regarding a "PoS" consensus. It was not easy to reach an stable consensus as we have nowadays and preserve 1 block finality, @erikzhang has been as mastermind and a good team has been cooperating for making what we current have reality.
I prefer to see everything just as a multi-agent system in which we should reach consensus.

In this sense, I prefer to leave these other aforementioned projects conduct their own vision. Thus, I strongly motivate all NEO contributors to keep contributing with our current vision, small, simply but efficient and working in practice.

**Comment by toghrulmaharramov** [comment]
@vncoelho A very arguable point regarding "pioneering". If you're talking about the generalized Proof-of-Stake consensus protocols, then Snow White and Ouroboros are way ahead simply because they are the only ones to have complete formal proofs (near-complete in case of Ouroboros and Ouroboros Praos). Unfortunately, both of those belong to a family of chain-based Proof-of-Stake protocols, rather than the family of BFT-based Proof-of-Stake protocols to which dBFT belongs (funnily enough, dBFT resembles SpinningBFT much more than it does PBFT, but I'll leave that argument for another day). Algorand have managed to create the first (what else do you expect when you have Turing Prize-winning Silvio Micali heading the research of the consensus protocol?) permission-less Proof-of-Stake protocol with instant finality while HoneyBadgerBFT is the first efficient asynchronous protocol with instant finality, while protocols such as LinBFT make use of threshold and BLS signatures to create two-phase protocols based on classic BFT (with no accidental forking, unlike dBFT 1.0), so I'm not sure what part of the Proof-of-Stake is NEO pioneering (especially considering the fact that the underlying Proof-of-Stake protocol of dBFT has no formal proofs whatsoever).

**Comment by vncoelho** [comment]
Turing Prize-winning Barbara Liskov is also one of the heading researcher of pBFT.

HoneyBadger is still a big question to me, @toghrulmaharramov. pBFT and HoneyBadger are articles that we read and discussed several times. The question between asynchronous and partially synchronous is an interesting discussion that I believe we gonna see the literature split for some more time.

I suggest these readings here:
* Scalable Byzantine Consensus via Hardware-assisted Secret Sharing, Jian Liu, Wenting Li,Ghassan O. Karame, Member, IEEE, and N. Asokan, Fellow, IEEE
* BEAT: Asynchronous BFT Made Practical
* Making Byzantine Fault Tolerant Systems Tolerate Byzantine Faults

**Comment by toghrulmaharramov** [comment]
@vncoelho not arguing with the fact that PBFT was revolutionary in 1999, however, it was designed for a different use case and with different computational power in mind (hence why the original paper proposed using MACs instead digital signatures).

In my opinion, HoneyBadger is a sound protocol with the only downside being the threshold signature scheme used and requirement of a relatively stable committee as a result (hence why it is permissioned).

Thanks for the suggestion, read all three of those papers. :D

**Comment by vncoelho** [comment]
I see, Tog, but science is made everyday...aehuaheau :+1:
I consider that the best and pioneer adjustment of the pBFT for consensus in Blockchain scenario is NEO dBFT, this is what I mentioned.

However, I still believe that other ones can still be good (or even better). But they still did not convinced us.

**Comment by shargon** [comment]
We should add a multiply factor inside policy native contract (1 by default). This could allow us to reduce the price without make a new release if the price go "to the moon"

**Comment by vncoelho** [comment]
Sounds interesting, @shargon, you mean a variable just for CN to increase (multiplying fees) for a given period?
I believe that we could even set another variable for limit this multiply until `X` number of blocks.

**Comment by lock9** [comment]
> We should add a multiply factor inside policy native contract (1 by default). This could allow us to reduce the price without make a new release if the price go "to the moon"

I think this is a great idea, I created an issue for you

**Comment by erikzhang** [comment]
@shargon Are you talking about `sys_fee` or `net_fee`?
